app/src/main/kotlin/com/wire/android/ui/calling/SharedCallingViewModel.kt

```diff
@@ -178,8 +188,24 @@ class SharedCallingViewModel @Inject constructor(
                 callState = callState.copy(
                     callStatus = call.status,
                     callerName = call.callerName,
-                    isMuted = call.isMuted,
                     isCameraOn = call.isCameraOn,
+                    isMuted = call.isMuted
+                )
+            }
+            true
+        }
+    }
+
+    private suspend fun observeParticipants(sharedFlow: SharedFlow<List<Call>>) {
+        sharedFlow.collect { calls ->
+            calls.find { call ->
+                call.conversationId == conversationId &&
+                        call.status != CallStatus.CLOSED &&
+                        call.status != CallStatus.MISSED
```

Comment:
If it's always filtered then maybe we could just make this `sharedFlow` filter the list once and emit only this single item - it's always a little less computations 😄

Reply:
Yeah, good point